Makefile

```
@@ -266,4 +293,41 @@ dev/down:
 .PHONY: test-e2e $(driver)
 test-e2e:
 	cd deploy; source ./../e2e/local-e2e.sh && run $(driver)
-	go test -v $(shell go list ./e2e)
+	$(GO) test -v $(shell $(GO) list ./e2e)
+
+.PHONY: $(DOCKER_BUILDER)
+$(DOCKER_BUILDER): Dockerfile.cross-builder | $(OUT_DIR) check_$(CMD_DOCKER)
+ 	# Build an image on top of goreleaser/goreleaser-cross:${GOLANG_CROSS_VERSION} with the necessary dependencies.
+	$(CMD_DOCKER) build -t $(DOCKER_BUILDER):$(GOLANG_CROSS_VERSION) --build-arg=GOLANG_CROSS_VERSION=$(GOLANG_CROSS_VERSION) - < $<
+
+# docker_builder_make runs a make command in the parca-agent-builder container
+define docker_builder_make
+	$(CMD_DOCKER) run --rm \
+	-v $(abspath $(DOCKER_BUILDER_KERN_SRC_MNT)):$(DOCKER_BUILDER_KERN_SRC_MNT) \
+	-v $(abspath .):/parca-agent/parca-agent \
+	-w /parca-agent/parca-agent \
+	--entrypoint make $(DOCKER_BUILDER) KERN_BLD_PATH=$(DOCKER_BUILDER_KERN_BLD) KERN_SRC_PATH=$(DOCKER_BUILDER_KERN_SRC) $(1)
+endef
+
+# test cross-compile release pipeline:
+.PHONY: release-dry-run
+release-dry-run: $(DOCKER_BUILDER) bpf
+	$(CMD_DOCKER) run \
+		--rm \
+		--privileged \
+		-v /var/run/docker.sock:/var/run/docker.sock \
+		-v `pwd`:/__w/parca-agent/parca-agent \
```

Nit variable instead of the command here (even if it is unlikely to fail, the command creates a sub-process, and its return code is hidden by the command it is nested in)

```suggestion
		-v "$(PWD):/__w/parca-agent/parca-agent" \
```